UNITTESTS/README.md

```diff
@@ -0,0 +1,357 @@
+# Unit testing Mbed OS
+
+This document describes the motivation for unit testing and how to run and write unit tests for Mbed OS.
+
+## Introduction
+
+Traditional software testing is defined into three main levels: unit testing, integration testing and system testing. These levels are often pictured as a pyramid to indicate the amount of testing per level.
+```
+^ Testing level
+|
+|       /\
+|      /  \       System testing
+|     /----\
+|    /      \     Integration testing
+|   /--------\
+|  /          \   Unit testing
+| /------------\
+|
+*-------------------> Amount of tests
+```
+Integration and system testing are performed in an environment with full Mbed OS included in the build.
+
+Unit testing takes place in a build environment where each C/C++ class or module is tested in isolation. This means test suites are built into separate test binaries and all access outside is stubbed to remove dependency of any specific hardware or software combination. This allows the testing to be done quickly using native compilers on the build machine.
+
+## Required tools
+
+* GNU toolchains installed.
+    * GCC 6 or later
+    * MinGW-W64 GCC-6.4.0 or MinGW-W64 GCC-7.3.0 (Windows)
+* CMake 3.0+ installed.
+* Python and pip installed.
```

Comment:
Which python versions we support? Need to define. And maybe pip also. Something like python 2.7.x or >3.5